src/Controller/Component/FlashComponent.php

```diff
@@ -100,7 +102,7 @@ public function set($message, array $options = [])
         }
 
         $messages = [];
-        if ($options['clear'] === false) {
+        if ($options['stacked'] === true && $options['clear'] === false) {
```

Comment:
If `stacked` is false value of `clear` becomes irrelevant and existing value (if any) won't be read.